das/stats.go

```diff
@@ -0,0 +1,30 @@
+package das
+
+// SamplingStats collects information about the DASer process. Currently, there are
+// only two sampling routines: the main sampling routine which performs sampling
+// over current network headers, and the `catchUp` routine which performs sampling
+// over past headers from the last sampled checkpoint.
+type SamplingStats struct {
+	// all headers before SampledBefore were successfully sampled
+	SampledBefore uint64 `json:"sampled_before_height"`
+	// MaxKnown is the height of the newest known header
+	MaxKnown uint64 `json:"max_known_height"`
+	// Failed contains all skipped header's heights with corresponding try count
+	Failed map[uint64]int `json:"failed,omitempty"`
+	// Workers has information about each currently running worker stats
+	Workers []WorkerStats `json:"workers,omitempty"`
+	// Concurrency currently running parallel workers
+	Concurrency int `json:"concurrency"`
+	// CatchUpDone indicates whether all known headers are sampled
+	CatchUpDone bool `json:"catch_up_done"`
+	// IsRunning tracks whether routine is running
+	IsRunning bool `json:"is_running"`
```

Comment:
This tracks only if the catch up process is running? We should make it explicit what `IsRunning` refers to. 

Also, the below statement will set IsRunning to `true` if the catchUp process is finished via `s.catchUpDone`. 
Only one of the below statements has to be true for IsRunning to be true. So if len(workers) == 0 BUT s.catchUpDone is true, then IsRunning will return true.
```
		IsRunning:     len(workers) > 0 || s.catchUpDone,
```

Reply:
IsRunning is meant to indicate that DASer is running. If `catchUpDone == true && len(workers) == 0` DASer is still subscribed to new headers and can resume sampling once new header is found. Maybe add some clarity to comment line like:
"IsRunning tracks whether routine is running and ready to sample new headers"

Reply:
Okay so `IsRunning tracks whether the DASer service is running'

Reply:
rephrased the  comment